### PR TITLE
[RFC] regression quick fixes

### PR DESCRIPTION
**Pull request by jberdine**
There are some regressions that I don't know the best fix that fits into the overall plans. I made this PR in order to get something usable, and to be more concrete than opening issues, but the changes themselves are probably not the eventually wanted ones. See the individual commit messages for more info on each.

### CODE REVIEW

**Comment by gpetiot** [comment]
Do you have some code samples that are impacted by 108b96e ? I removed this code since it looked redundant with the generic treatment of sequences and test_branch didn't show any changes.

**Comment by jberdine** [comment]
I wonder if you used a version of infer that is reasonably recent when running test_branch. There are a number of changes, for example around code of the form `[%Trace.call ...] ; ...`.

File: src/Fmt_ast.ml [hunk 1]
```diff
@@ -906,8 +906,8 @@ and fmt_pattern c ?pro ?parens ({ctx= ctx0; ast= pat} as xpat) =
           $ fmt_or_k last p.sep_after_final p.sep_after_non_final
         in
         hvbox 0
-          (Cmts.fmt c ppat_loc
-             (p.box
+          (p.box
+             (Cmts.fmt c ppat_loc
```
Comment (Julow) [inline src/Fmt_ast.ml]: This patch cause an unstable comment in `test/passing/comments.ml`.

Comment (jberdine) [reply src/Fmt_ast.ml]: FYI, this commit is separated out in https://github.com/ocaml-ppx/ocamlformat/pull/1141

Do you know if that test was stable before https://github.com/ocaml-ppx/ocamlformat/pull/866 ? I don't know if there is some intended interaction between the other changes in that PR and this one, or if this change was just added to that PR by mistake.

File: src/Fmt_ast.ml [hunk 2]
```diff
@@ -1697,7 +1698,12 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
                              | Some i when i <= 2 -> "@ "
                              | _ -> "@;<1 2>" ) )
                        $ fmt_expression c ?box xbody )
-                   $ str ")" $ Cmts.fmt_after c pexp_loc )
+                   $ fmt_or_k
+                       Poly.(c.conf.indicate_multiline_delimiters = `Space)
+                       (fmt_or_k fit (str ")")
+                          (fits_breaks ~force_fit_if:fit ")" ~hint:(1, 0) ")"))
+                       (fits_breaks ~force_fit_if:fit ")" ~hint:(0, 0) ")")
```
Comment (Julow) [inline src/Fmt_ast.ml]: I don't remember why this parentheses are gone. @gpetiot what do you think ?
Could it be that there was no space after the opening ?
eg, it formats like this:
```
   foo (fun () ->
       ...
       abc )

  foo
     (let a = ... in
      a)
```

Comment (jberdine) [reply src/Fmt_ast.ml]: In such cases the opening space was originally taken to be one following `(fun`.

Comment (gpetiot) [reply src/Fmt_ast.ml]: I don't remember either, and I think it was decided it was okay to sometimes have no space at the beginning but to have a space at the end, especially for `(fun` even if `indicate-multiline-delimiters=space`. I didn't manage to find the issue/PR where this have been discussed. There have been a lot of PRs to improve the consistency of parentheses, but maybe some regressions have been introduced along the way. I'm favorable to improve the consistency anyway.

File: src/Fmt_ast.ml (hunk 1 above)
Comment (Julow) [reply src/Fmt_ast.ml]: I think before that PR, it was a dropped comment. The test is added by this PR.

Comment (jberdine) [reply src/Fmt_ast.ml]: Ok. The high-level question is if, knowing the details of 866, you know of a way to not drop this comment without the regression on the example in https://github.com/ocaml-ppx/ocamlformat/issues/1140 ?

File: src/Fmt_ast.ml (hunk 2 above)
Comment (Julow) [reply src/Fmt_ast.ml]: I moved that commit to a PR: https://github.com/ocaml-ppx/ocamlformat/pull/1169

**Comment by gpetiot** [comment]
@jberdine can you rebase your PR so we can have a clearer view? I think a lot of things are now fixed in master.

**Comment by jberdine** [comment]
These changes are now covered by fixes in master, https://github.com/ocaml-ppx/ocamlformat/pull/1169, https://github.com/ocaml-ppx/ocamlformat/pull/1342, and https://github.com/ocaml-ppx/ocamlformat/pull/1343.

**Pull request closed without merging**